backends/platform/ios7/ios7_gamepad_controller.mm

```diff
@@ -162,10 +166,29 @@ - (void)controllerDidConnect:(NSNotification *)notification {
 
 		_controller.extendedGamepad.dpad.valueChangedHandler = ^(GCControllerDirectionPad * _Nonnull dpad, float xValue, float yValue) {
 			// Negative values are left/down, positive are right/up, 0 is no press
-			[self handleJoystickButtonAction:Common::JOYSTICK_BUTTON_DPAD_LEFT isPressed:(xValue < 0)];
-			[self handleJoystickButtonAction:Common::JOYSTICK_BUTTON_DPAD_RIGHT isPressed:(xValue > 0)];
-			[self handleJoystickButtonAction:Common::JOYSTICK_BUTTON_DPAD_UP isPressed:(yValue > 0)];
-			[self handleJoystickButtonAction:Common::JOYSTICK_BUTTON_DPAD_DOWN isPressed:(yValue < 0)];
+			if (xValue != currentDpadXValue) {
+				if (xValue < 0) {
+					[self handleJoystickButtonAction:Common::JOYSTICK_BUTTON_DPAD_LEFT isPressed:true];
+				} else if (xValue > 0) {
+					[self handleJoystickButtonAction:Common::JOYSTICK_BUTTON_DPAD_RIGHT isPressed:true];
+				} else if (xValue == 0 && currentDpadXValue < 0) {
+					[self handleJoystickButtonAction:Common::JOYSTICK_BUTTON_DPAD_LEFT isPressed:false];
+				} else {
+					[self handleJoystickButtonAction:Common::JOYSTICK_BUTTON_DPAD_RIGHT isPressed:false];
+				}
+				currentDpadXValue = xValue;
+			} else if (yValue != currentDpadYValue) {
```

Comment:
With this approach you can get down events with no corresponding up events. Would that not be an issue?

Would it not be better to do something like:
```c++
// Negative values are left/down, positive are right/up, 0 is no press
// Change value to only be -1, 0, or 1
xValue = xValue < 0.f ? -1.f : xValue > 0.f ? 1.f : 0.f;
if (xValue != currentDpadXValue) {
	if (currentDpadXValue < 0) {
		[self handleJoystickButtonAction:Common::JOYSTICK_BUTTON_DPAD_LEFT isPressed:false];
	} else if (currentDpadXValue > 0) {
		[self handleJoystickButtonAction:Common::JOYSTICK_BUTTON_DPAD_RIGHT isPressed:false];
	}
	if (xValue < 0) {
		[self handleJoystickButtonAction:Common::JOYSTICK_BUTTON_DPAD_LEFT isPressed:true];
	} else if (xValue > 0) {
		[self handleJoystickButtonAction:Common::JOYSTICK_BUTTON_DPAD_RIGHT isPressed:true];
	}
}
// And same for Y
[...]
```

Reply:
Yes, I'll update

Reply:
Was this updated? It seems to me that the issue is still present in the current code. Or with this not an issue after all?

Reply:
Oh, I forgot to push the last commit I had which had the changes. Sorry for that. I'll push it tomorrow.